.github/workflows/release.yml

```diff
@@ -71,18 +41,39 @@ jobs:
       - name: Install dependencies
         run: |
           apt-get update
-          apt-get install -y build-essential cmake gettext locales ninja-build pkg-config unzip
+          apt-get install -y build-essential cmake gettext ninja-build unzip
       - if: github.event_name == 'push' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name != 'nightly')
         run: CC=gcc-10 make appimage-latest
       - if: github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name == 'nightly')
         run: CC=gcc-10 make appimage-nightly
+      - if: github.event_name == 'push' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name != 'nightly')
+        run: printf 'NVIM_BUILD_TYPE=Release\n' >> $GITHUB_ENV
+      - if: github.event_name == 'schedule' || (github.event_name == 'workflow_dispatch' && github.event.inputs.tag_name == 'nightly')
+        run: printf 'NVIM_BUILD_TYPE=RelWithDebInfo\n' >> $GITHUB_ENV
```

Comment:
Out of interest, why this conditional? Why not always build `RelWithDebInfo`?

Reply:
I couldn't decide whether defaulting to RelWithDebInfo like the appimage, or adjust the appimage to build on Release for releases and RelWithDebInfo otherwise. I think the latter is more consistent since that's what every other release artifact does, but it'd require a bit more work that I thought would be more fitting in a future PR. In that case it shouldn't be removed. Basically: I'm unsure if this is needed in the future, so I erred on the side of caution. I don't mind removing it though.

Reply:
I think that if in doubt, the appimage is the one to stick with -- that's where the most deliberation went into. And consistency is important here; the goal is to be able to rely on some assumptions (like build type here), the better. So my vote would be to make _every_ release (at least on Linux) `RelWithDebInfo`.

(macOS and Windows are different, since it's not as easy to get stacktraces there anyway.)

Reply:
But just a thought; I'll defer to you and @bfredl on this.

Reply:
Eh, sure, I don't mind. I'll default to RelWithDebInfo. All of this is going to need a serious once-over anyway.

Reply:
Give me a minute, maybe I can adjust the appimage without running into too much problems. I think that'll be more consistent overall.

Reply:
Sure, sounds good. In any case, this needs a serious overhaul regardless as this is not maintainable in its current state.

Reply:
I think I can fix this for appimage as well. Let me take a look